src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            addIfFirstNode(newNode);
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexValidation(index, size + 1);
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
```

Comment:
isEmpty it the same that index == size so you can just call method add(T value) 
and [Don’t complicate if-else construction. ](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-complicate-if-else-construction-detailed-explanation)